### ISSUE TITLE
Inner types of variants aren't type-checked correctly

### ISSUE THREAD

**Issue by hulthe** [algebraicdb/algebraicdb#52]
```sql
CREATE TYPE MaybeInt AS VARIANT {
	Just(Integer),
	Nothing(),
};

CREATE TABLE t(m MaybeInt);

INSERT INTO t(m) VALUES (MaybeInt::Just(42.42));
```

This passes the typechecker, even though `MaybeInt::Just(42.42)` should be a type-error.

**Comment by hulthe** [comment]
We could and should probably solve this while when we implement duck-typing

**Comment by hulthe** [comment]
Solved by #54
